src/ui/React/CompanyDropdown.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * Creates a dropdown (select HTML element) with company names as options
+ */
+import React from "react";
+import { companiesMetadata } from "../../Company/data/CompaniesMetadata";
+
+import Select, { SelectChangeEvent } from "@mui/material/Select";
+import MenuItem from "@mui/material/MenuItem";
+import Button from "@mui/material/Button";
+
+interface IProps {
+  purchase: () => void;
+  canPurchase: boolean;
+  onChange: (event: SelectChangeEvent<string>) => void;
+  value: string;
+}
+
+export function CompanyDropdown(props: IProps): React.ReactElement {
+  const companies = [];
+  for (const company of companiesMetadata.sort((a, b) => a.name.localeCompare(b.name))) {
```

Comment:
Basically this means that every time the CompanyDropdown react element is rerendered, it will try to sort the entire companiesMetadata array again.
Could instead sort it in the file where it's initialized, and then there is no need to sort it again later.

Reply:
Fixed.

Reply:
Note that ServerDropdown does the same thing, fix should also be applied there at some point.